docs/docs/configuration/providers/azure.md

```diff
@@ -37,6 +37,13 @@ title: Azure
    --oidc-issuer-url=https://login.microsoftonline.com/{tenant-id}/v2.0
 ```
 
+
+***Roles***:
+
+Microsoft Entra ID (Former Azure AD) adds a `roles` claim when App Roles are defined in the App registrations and assigned to the logged in users. Microsoft Entra ID roles can be autorized usign the `--allowed-role=<role name>` option, allowing only users from the defined list of roles (non-exclusive).
```

Comment:
```suggestion
Microsoft Entra ID (Former Azure AD) adds a `roles` claim when App Roles are defined in the App registrations and assigned to the logged in users. Microsoft Entra ID roles can be authorized using the `--allowed-role=<role name>` option, allowing only users from the defined list of roles (non-exclusive).
```

Reply:
Hi @tuunit!

No worries, I hope you are feeling better and happy new year 😄 . 

In our use case, we have 4 different groups for our application, one for each environment (DEV, TST, VAL and PRD):

* AppUsers-DEV
* AppUsers-TST
* AppUsers-VAL
* AppUsers-PRD

These groups exist in Azure AD and are linked to our enterprise application for OIDC purposes (1 application per env). Inside each group, we are defining 4 roles that are called the same in each application:

* Viewer
* Editor
* Admin
* APIConsumer (This role is why we need the allowed roles feature)

If for example we wanted to grant admin permissions only on DEV but not in the rest of the environments, we assign the role Admin in the AppUsers-DEV enterprise app.

Problem comes when we want to separate APIConsumers from the rest of the users, but keeping the same Enterprise App. Without this validation, any user that has a valid user in the group (no matter the role) can make calls to the API in the backend (A Viewer making POST calls is something we don't want to allow).

Unfortunately, we don't own the backend API and we can't make this authorization work in any other way but with some middleware in between like OAuth2 proxy. We have considered other options like a JWT validator, but it only works when we use JWT tokens for the authentication, and that would add yet more complexity to the stack.

I hope the explanation is clear, and if you have any other ideas on how to make this setup work, we would be more than happy to hear them 😄

Reply:
Thank you for the detailed explanation :) Hope you had a great start into the New Year.